### ISSUE TITLE
Improve "Your First Tuari App" page as it is (at least) lacking the "Build" step [docs] 

### ISSUE THREAD

**Issue by timfayz** [tauri-apps/tauri#3987]
Recently, I stuck upon Tauri and I found it very promising after reading its first pages. I needed exactly what it promises – "simple", fast, and small desktop application with a web frontend. So I decided to give it a try. My goal was "just" to build a basic Hello World application, given the tutorial a normal newbie would start from: https://tauri.studio/docs/getting-started/beginning-tutorial/. After an hour, I realized that the promises were good but the "end user" experience of actual building an app is just horrible (sorry but I don't have any better words to describe it). I thought that I do something wrong or was reading the wrong page.. However, after I watched a video of someone else did the same thing, namely, building his first app following that page, I realized that I'm not the only one. Here is the key moment that is 1:1 corresponds to me: https://www.youtube.com/watch?v=AsynWNbU0X0&t=1492s. Besides other confusion parts (if you rewind back) it shows how people are struggling when they are trying to understand how to build things with Tauri.

If you aren't familiar with rust / Tauri infrastructure, there are other confusing things in that page (and in the video you can see that confusion as well):

1. [Install Tauri CLI package as a dev dependency](https://tauri.studio/docs/getting-started/beginning-tutorial/#install-tauri-cli-package-as-a-dev-dependency).  What is 'local package' is supposed to mean in "If you decide to use Tauri as a local package ..."? (only while writing it, now I actually realized)
1. [Tauri CLI as a cargo subcommand](https://tauri.studio/docs/getting-started/beginning-tutorial/#alternatively-install-tauri-cli-as-a-cargo-subcommand). What is "Tauri CLI"?
1. [Install Tauri API Package as a Dependency](https://tauri.studio/docs/getting-started/beginning-tutorial/#install-tauri-api-package-as-a-dependency-optional) What is "Tauri API"?

Basically, before introducing any fancy words and modules like Tauri API, CLI, etc that are specific to the infrastructure that is yet unknown, you'd better to give a short description on what it means and why should I care.

I hope you will watch the video and will understand that no matter how good your project is without adequate (not even good) documentation is just useless. After such a pain, I'm certainly switch to Electron as it doesn't cost that pain. Hope that helps you to improve. Thank you for your effort!

**Comment by allynsweet** [comment]
I'm new to using Tauri as well, and new technologies take some learning, but as a user of software, my computer's memory encourages you to stick with it a little while longer before jumping to a Chromium-based framework.

> What is 'local package' is supposed to mean in "If you decide to use Tauri as a local package ..."?

Basically adding Tauri to your project's package.json or globally on your operating system. `yarn add -D @tauri-apps/cli` will add the dev dependency to your project's package.json while `yarn global add @tauri-apps/cli` will add it to your operating system. It's better to add it locally for better versioning control, and the local CLI can be called using `yarn tauri` while the global CLI can be used using just `tauri`.

> What is "Tauri CLI"?

The command line interface for interfacing with Tauri via terminal. The docs on this could be found under the API -> CLI section of the website [here](https://tauri.studio/docs/api/cli). I agree this should be hyperlinked from the Your First Tauri App section.

> What is "Tauri API"?

The way your frontend interfaces with the Rust. Similar to Electron's API that is used between frontend and NodeJS. The full docs for this for JS/TS can be found [here](https://tauri.studio/docs/api/js/), and Rust can be found [here](https://docs.rs/tauri/1.0.0-rc.0/tauri/).

The "Your First Tauri App" section is really about setting up the project, agnostic to any framework you use (Vue, React, etc). Take a look at [Tauri Awesome](https://github.com/tauri-apps/awesome-tauri) to get an idea of how to set up a boilerplate project for your chosen framework.

**Comment by timfayz** [comment]
@FabianLars, it would be nice to see the drafts if they are available. Can hardly relate the pieces in the current version. Honestly, I don't know who wrote the current version but this the worst documentation I've seen so far in projects with such a big amount of stars (is that a 'star fever'?). I want to mention good examples of how "Build your first app" should look like. Just borrow and replace! No one would complain! ;)

- https://neutralino.js.org/docs/getting-started/your-first-neutralinojs-app (at least how to build the project isn't missed completely)
https://www.electronjs.org/docs/latest/tutorial/quick-start (very nice and detailed explanation on every step with a good sum up at the end [in case you're lost anyway])

@allynsweet Thank you for such a comprehensive reply. It should help the purpose. I'll try it tomorrow again.

> The "Your First Tauri App" section is really about setting up the project, agnostic to any framework you use...

Well...this is exactly what I needed – setting up and running my first Tauri app _without_ frameworks obsession. And this is exactly the place where the special attention needs to be paid in helping people to go through details (surely without messing around with complicated frameworks on top of the already vague docs).

Also, I didn't find information about "project structure". Is there any? As for now the only thing that comes to my mind is "Why there is such a clatter? I hope you don't expect people just to ignore it, right?" `.rustc_info.json`, `tauri.conf.json`, `build.rs`, `Cargo.toml`, `CACHEDIR.TAG`, God.. 🤯  Feels like a complete mess!

![image](https://user-images.githubusercontent.com/6209197/165599856-251f0688-cdd0-470b-89de-9c90410b646d.png)

Just for comparison, attaching the one of Neutralinojs:

![image](https://user-images.githubusercontent.com/6209197/165601900-dc47ee26-5f08-4c68-892a-74fef80a185d.png)

I know, Neutralinojs is probably far from being your 'real' competitive; yet it found its place among the others.

**Comment by JonasKruckenberg** [comment]
First off, while your comments and feedback are definitely very helpful and well-meaning they feel unnecessarily aggressive. We're all just volunteers, no one works on this full time and we've been all focusing on stability of the framework itself. 
We're really doing our best here, so just be patient please.

Anyway, [here is the WIP version](https://github.com/JonasKruckenberg/tauri-docs-wip) of the new docs, no Getting-Started pages yet, but I'm working on these right as I write this.

**Comment by timfayz** [comment]
> First off, while your comments and feedback are definitely very helpful and well-meaning they feel unnecessarily aggressive.

Surely they are! Sorry for that. Didn't want to mean it 🙏 😇  (Verzeiht mir!)

> We're all just volunteers, ...

Sure. I would say it is just about being (extremely) pity that you've done such a great job but spend so little time on expanding the things that would take an hour for those who know but for those who don't – would take the days. 

> But all files in the target folder are normal files generated by rust/cargo that are generated during compilation. Nothing to worry about and nothing we can change.

That's very good to know! That information will actually motivate me to take a couple of tutorials on Rust and make myself comfortable with those! :) Once again, thank you a lot for the precious info. 

Btw, if I got a question where can I ask it? Stackoverflow with `tauri` tag? Probably polluting doc issues here isn't the best way to go...🥺

**Comment by JonasKruckenberg** [comment]
> Btw, if I got a question where can I ask it? Stackoverflow with `tauri` tag? Probably polluting doc issues here isn't the best way to go...🥺

Our discord is where we're most active: https://discord.com/invite/tauri

Otherwise Github discussion is also available. I personally don't have a Stackoverflow account nor do I know that any of the team has one, but I guess your question would still be answered there too.
